week-3/stretch/rotate-char.js

```diff
@@ -6,6 +6,33 @@
 // This function is commonly used for weak text encryption and decryption,
 // where shifting characters by a certain value can obscure their meaning or reveal hidden messages.
 
+function isLetterOrNumber(char) {
+  return /^[a-zA-Z0-9]+$/.test(char);
+}
+
+function rotateCharacter(char, shift) {
+  if (isLetterOrNumber(char) && char.length == 1 && typeof shift == "number" && shift >= 0) {
+    if (char.charCodeAt() > 47 && char.charCodeAt() < 58) {
+      return char;
+    } else {
+      const isUpperCase = /[A-Z]/.test(char);
```

Comment:
I haven't tested, but if this works, then cool. Working code is already great. I think the general principle that you are trying to deal with here is that you need to rotate within the lower case and upper case alphabets independently i.e. lower case always rotates to another lower case etc. and that is a large part of why it's a bit lengthy

You could probably do this a little more 'cheaply'. If you Google "caesar cipher javascript" you might get some ideas. But fundamentally, assuming this works correctly, good job :)

Reply:
Thanks, Dominic.
I will give a look at this asap.